### PR TITLE
节点按延迟排序；删除重复节点

### PR DESCRIPTION
**Pull request by banjiaojuhao**
对于节点比较多的情况还是需要这两个功能的。

### CODE REVIEW

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/ui/MainActivity.kt [hunk 1]
```diff
@@ -239,6 +239,11 @@ class MainActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedList
             true
         }
 
+        R.id.sort_all -> {
```
Comment (rurirei) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/ui/MainActivity.kt:247]: do you done a performance test, do that blocks activity?

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -65,6 +65,21 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
         }
     }
 
+    fun sortAllConfiguration() {
+        AngConfigManager.configs.vmess.sortBy {
+            val delay = it.testResult.substringBefore("ms").toIntOrNull() ?: 0
+            if (delay < 1) {
+                Int.MAX_VALUE
```
Comment (rurirei) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/viewmodel/MainViewModel.kt:74]: do warning msgs.

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/viewmodel/MainViewModel.kt [hunk 3]
```diff
@@ -65,6 +65,21 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
...
+            } else {
+                delay
+            }
+        }
+        AngConfigManager.storeConfigFile()
```
Comment (rurirei) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/viewmodel/MainViewModel.kt:79]: waited #865 (#701 in too.)

**Review by rurirei — CHANGES_REQUESTED** [review]

**Comment by Animisml** [comment]
请问有谁可以发个节点过来

**Pull request closed without merging**